cpp/src/arrow/compute/exec.h

```diff
@@ -47,6 +47,7 @@ class CpuInfo;
 
 namespace compute {
 
+struct FunctionExecutor;
 class FunctionOptions;
 class FunctionRegistry;
```

Comment:
Can we move these forward-decls to `compute/type_fwd.h` where they should belong?